api/db.py

```diff
@@ -42,6 +42,14 @@ async def find_by_id(self, model, obj_id):
         obj = await col.find_one({'_id': ObjectId(obj_id)})
         return model(**obj) if obj else None
 
+    async def find_by_attribute(self, model, query_list):
```

Comment:
```suggestion
    async def find_by_attribute(self, model, attributes):
```

Reply:
The argument passed from the query is a dictionary with attribute key/value pairs, not a list.  So I think `attributes` is a better name, also we should be adding docstrings to clarify how this works.